### PR TITLE
[feat]: Network resources created using public and private subnets, r…

### PR DESCRIPTION
**Pull request by leohauschild**
…outing with internet gw and NAT

### CODE REVIEW

File: aws-environments/infrastructure/internet-gw.tf [hunk 1]
```diff
@@ -0,0 +1,22 @@
+resource "aws_internet_gateway" "gw" {
+  vpc_id = aws_vpc.project_vpc.id
+
+  tags = {
+    Name = format("gw-%s", var.project-name)
```
Comment (carlosgit2016) [inline aws-environments/infrastructure/internet-gw.tf:5]:
```suggestion
    Name = "gw-${var.project-name}"
```

File: aws-environments/infrastructure/versions.tf [hunk 2]
```diff
@@ -0,0 +1,26 @@
+terraform {
+  required_version = "~> 1.5.3"
```
Comment (carlosgit2016) [inline aws-environments/infrastructure/versions.tf]:
```suggestion
  required_version = ">= 1.5.3, < 2.0.0"
```

File: aws-environments/infrastructure/variables.tf [hunk 3]
```diff
@@ -0,0 +1,5 @@
+variable "project-name" {
```
Comment (carlosgit2016) [inline aws-environments/infrastructure/variables.tf]: Up to you, but usually the community prefer using underscore
```suggestion
variable "project_name" {
```

File: aws-environments/infrastructure/subnet-public.tf [hunk 4]
```diff
@@ -0,0 +1,33 @@
+resource "aws_subnet" "public_subnet_1a" {
```
Comment (carlosgit2016) [inline aws-environments/infrastructure/subnet-public.tf:1]: Im curious why 4 subnets, 2 public and 2 privates

File: aws-environments/infrastructure/nat-gateway.tf [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+resource "aws_nat_gateway" "nat_gw" {
+  allocation_id = aws_eip.vpc_eip.id
+  subnet_id     = aws_subnet.public_subnet_1a.id
+
+  tags = {
+    Name = format("nat-gw-%s", var.project-name)
+  }
+}
+
+resource "aws_route_table" "nat_route_table" {
```
Comment (carlosgit2016) [inline aws-environments/infrastructure/nat-gateway.tf:18]: That's up to you, but I think 2 route table is overhead to the same VPC, the igw and nat route can be in the same

File: aws-environments/infrastructure/internet-gw.tf (hunk 1 above)
Comment (benhur-araujo) [reply aws-environments/infrastructure/internet-gw.tf:5]: @carlosgit2016 did you suggest changing it to simply the code or for some other reason?

File: aws-environments/infrastructure/subnet-public.tf (hunk 4 above)
Comment (leohauschild) [reply aws-environments/infrastructure/subnet-public.tf:1]: Explained in discord conversation

**Pull request merged**